common/primitives/src/msa.rs

```diff
@@ -62,6 +72,16 @@ pub struct SchemaGrant<SchemaId, BlockNumber> {
 	pub revoked_at: BlockNumber,
 }
 
+impl<SchemaId, BlockNumber> PartialEq for DelegationResponse<SchemaId, BlockNumber>
+where
+	SchemaId: PartialEq,
+	BlockNumber: PartialEq,
+{
+	fn eq(&self, other: &Self) -> bool {
+		self.provider_id == other.provider_id && self.permissions == other.permissions
```

Comment:
Does the Rust `==` operator on `Vec` test equality of the elements in a Set-like fashion? I didn't think it did... if not, do we want something like that?

Reply:
It works, since if the inner type doesn't include the partialEq or Eq it doesn't compile.